### PR TITLE
speed up `validatorDb`/`validatorPerf`

### PR DESCRIPTION
**Pull request by arnetheduck**
avoid needless hashing and use bigger txn:s

### CODE REVIEW

File: ncli/ncli_db.nim [hunk 1]
```diff
@@ -746,14 +753,20 @@ proc cmdValidatorDb(conf: DbConf, cfg: RuntimeConfig) =
           int64(target_attester), # Target delta
           int64(head_attester), # Head delta
           delay)).expect("DB")
-    outDb.exec("COMMIT;").expect("DB")
+    if getStateField(state[].data, slot).epoch.int64 mod 16 == 0:
```
Comment (tersec) [inline ncli/ncli_db.nim:756]: Seems potentially more elegant to use something like 16'u64 or some other type-compatible constant than to convert the runtime-variable `getStateField(state[].data, slot).epoch` to a theoretically incompatible integer type.

File: ncli/ncli_db.nim [hunk 2]
```diff
@@ -775,6 +788,10 @@ proc cmdValidatorDb(conf: DbConf, cfg: RuntimeConfig) =
     if getStateField(state[].data, slot).isEpoch():
       processEpoch()
 
+  if inTxn:
+    inTxn = false
+    outDb.exec("COMMIT;").expect("DB")
```
Comment (tersec) [inline ncli/ncli_db.nim:793]: How much does this commit-every-16-epochs optimization help, quantitatively?

**Review by tersec — APPROVED** [review]

File: ncli/ncli_db.nim (hunk 2 above)
Comment (arnetheduck) [reply ncli/ncli_db.nim:793]: that'll be highly dependent on the drive, but about 2x:ish on my laptop - what happens on commit is more or less that the WAL gets moved to the real db and an fsync is made.

File: ncli/ncli_db.nim (hunk 1 above)
Comment (arnetheduck) [reply ncli/ncli_db.nim:756]: yeah I thought about that but we already have to convert to int64 to insert row into database (no uint64 in sqlite).. it's a bridge to cross if ever it crashes

**Pull request merged**